tomopy/prep/alignment.py

```diff
@@ -443,3 +455,240 @@ def shift_images(prj, sx, sy):
     prj *= scl
 
     return prj
+
+
+def find_slits_corners_aps_1id(img, 
+                               method='quadrant+', 
+                               medfilt2_kernel_size=3,
+                               medfilt_kernel_size=23,
+                               ):
+    """
+    Automatically locate the slit box location by its four corners.
+
+    Parameters
+    ----------
+    img                   :  np.2darray
+        input 2D images
+    method                :  str,  ['simple', 'quadrant', 'quadrant+'], optional
+        method for auto detecting slit corners
+    medfilt2_kernel_size  :  int, optional 
+        2D median filter kernel size for noise reduction
+    medfilt_kernel_size   :  int, optional 
+        1D median filter kernel size for noise reduction
+    
+    Returns
+    -------
+    tuple
+        autodetected slit corners (counter-clockwise order)
+        (upperLeft, lowerLeft, lowerRight, upperRight)
+    """
+
+
+    img = medfilt2d(np.log(img.astype(np.float64)),
+                    kernel_size=medfilt2_kernel_size,
+                    )
+    rows, cols = img.shape
+    
+    # simple method is simple, therefore it stands out
+    if method.lower() == 'simple':        

```

Comment:
I'd refactor this if else conditions by creating and calling simple, quadrant and quadrant+ functions here.

This would also give a chance to explain the purpose of each in their own function docs.

Reply:
The functionality of each method is added to the docstring.  

As for refactoring, I have to argue that breaking them into separate functions would make the code less readable and difficult to maintain.

Reply:
I think now I agree with you. I assume no other function will call these functions, so they better stay where they are right now.